onnx/test/runtime_backend_test.py

```diff
@@ -0,0 +1,632 @@
+                    )
+        elif isinstance(desired, list):
+            if not isinstance(output, list):
+                raise AssertionError(
+                    f"Expected result is 'list' but output type is {type(output)} for output {i_output}"
+                    f"\n--EXPECTED--\n{desired}\n--GOT--\n{output}."
+                )
+            if len(desired) != len(output):
+                raise AssertionError(
+                    f"Expected has {len(desired)} but output has {len(output)} for output {i_output}"
+                    f"\n--EXPECTED--\n{desired}\n--GOT--\n{output}."
+                )
+            for a, b in zip(desired, output):
+                self._compare_results(index, i_output, a, b, rtol=rtol, atol=atol)
+        else:
+            raise NotImplementedError(
+                f"Comparison not implemented for type {type(desired)} and output {i_output}."
+            )
+
+    def is_random(self):
+        "Tells if a test is random or not."
+        if "bernoulli" in self.folder:
+            return True
+        return False
+
+    def run(self, load_fct, run_fct, index=None, rtol=1e-07, atol=0):
+        """
+        Executes a tests or all tests if index is None.
+        The function crashes if the tests fails.
+
+        :param load_fct: loading function, takes a loaded onnx graph,
+            and returns an object
+        :param run_fct: running function, takes the result of previous
+            function, the inputs, and returns the outputs
+        :param index: index of the test to run or all.
+        :param rtol: relative tolerance
+        :param atol: absolute tolerance
+        """
+        if index is None:
+            for i in range(len(self)):
+                self.run(load_fct, run_fct, index=i, atol=atol, rtol=rtol)
+            return
+
+        obj = load_fct(self.onnx_model)
+
+        got = run_fct(obj, *self.tests[index]["inputs"])
+        expected = self.tests[index]["outputs"]
+        if len(got) != len(expected):
+            raise AssertionError(
+                f"Unexpected number of output (test {index}, folder {self.folder!r}), "
+                f"got {len(got)}, expected {len(expected)}."
+            )
+        for i, (e, o) in enumerate(zip(expected, got)):
+            if self.is_random():
+                if e.dtype != o.dtype:
+                    raise AssertionError(
+                        f"Output {i} of test {index} in folder {self.folder!r} failed "
+                        f"(type mismatch {e.dtype} != {o.dtype!r})."
+                    )
+                if e.shape != o.shape:
+                    raise AssertionError(
+                        f"Output {i} of test {index} in folder {self.folder!r} failed "
+                        f"(shape mismatch {e.shape} != {o.shape})."
+                    )
+            else:
+                self._compare_results(index, i, e, o, atol=atol, rtol=rtol)
+
+
+def enumerate_onnx_tests(series, fct_filter=None):
```

Comment:
not to local from file system, utilize onnx.backend.test.case.node.collect_testcases()